spec/support/webpacker.rb

```diff
@@ -0,0 +1,36 @@
+module WebpackerTest
+  TIMESTAMP_FILE = Rails.root.join("tmp", "webpacker-#{Rails.env}-timestamp")
+
+  def self.compile_once
+    digest_file = Rails.root.join("tmp/webpacker_#{Rails.env}_digest")
+
+    app_code = Dir[Webpacker.config.source_path.join("**/*")]
+    npm_code = Dir[Rails.root.join("yarn.lock")]
+
+    packable_contents = (app_code + npm_code)
+      .sort
+      .map { |filename| File.read(filename) if File.file?(filename) }
+      .join
+    digest = Digest::SHA256.hexdigest(packable_contents)
```

Comment:
Hopefully I am understanding the question correctly :-) ...

The reason for checking if the digest has changed is to take into account any changes to assets since the digest was last created. If we do not check if the digest had changed then we would never recompile. At the same time we only want one process to compile or recompile if required.
This problem is perhaps more pertinent on CI as we will will always have to compile from scratch. It appears weird issues occur if you have multiple processes compiling the same files at the same time. This is what we want to avoid.

Reply:
My point is that if there are _no_ changes in the assets, `Webpacker.compile` doesn't do anything, as evidenced by the fact that it returns in less than 1/10th of a second (see benchmarks). If something is changed in the assets, I tested by adding some whitespace to one of the JS files, then `Webpacker.compile` takes 10s to, presumably, do the compilation.

So I wonder if all that code to check if anything has changed is actually necessary.